test/controllers/update/update.review.statement.controller.spec.ts

```diff
@@ -0,0 +1,130 @@
+
+import { authentication } from "../../../src/middleware/authentication.middleware";
+import { companyAuthentication } from "../../../src/middleware/company.authentication.middleware";
+import { postTransaction, closeTransaction } from "../../../src/service/transaction.service";
+import { updateOverseasEntity } from "../../../src/service/overseas.entities.service";
+import { startPaymentsSession } from "../../../src/service/payment.service";
+import { getApplicationData } from "../../../src/utils/application.data";
+import { isActiveFeature } from "../../../src/utils/feature.flag";
+import { APPLICATION_DATA_CH_REF_UPDATE_MOCK, APPLICATION_DATA_UPDATE_BO_MOCK, ERROR, OVERSEAS_ENTITY_ID, PAYMENT_LINK_JOURNEY, TRANSACTION_CLOSED_RESPONSE, TRANSACTION_ID } from "../../__mocks__/session.mock";
+import { UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL, UPDATE_PRESENTER_CHANGE_EMAIL, UPDATE_PRESENTER_CHANGE_FULL_NAME, UPDATE_REVIEW_STATEMENT_BEFORE_SUBMITTING_URL, UPDATE_REVIEW_STATEMENT_PAGE } from "../../../src/config";
+import { ANY_MESSAGE_ERROR, NO_CHANGE_REVIEW_STATEMENT_BENEFICIAL_OWNER_STATEMENT, NO_CHANGE_REVIEW_STATEMENT_BENEFICIAL_OWNER_STATEMENTS_CEASED_TITLE, NO_CHANGE_REVIEW_STATEMENT_PAGE_TITLE, NO_CHANGE_REVIEW_STATEMENT_WHO_CAN_WE_CONTACT, SERVICE_UNAVAILABLE, UPDATE_CHECK_YOUR_ANSWERS_CONTACT_DETAILS } from "../../__mocks__/text.mock";
+import { OverseasEntityKey, Transactionkey } from "../../../src/model/data.types.model";
+
+const mockIsActiveFeature = isActiveFeature as jest.Mock;
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockLoggerDebugRequest = logger.debugRequest as jest.Mock;
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockCompanyAuthenticationMiddleware = companyAuthentication as jest.Mock;
+mockCompanyAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
+mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockTransactionService = postTransaction as jest.Mock;
+mockTransactionService.mockReturnValue( TRANSACTION_ID );
+
+const mockOverseasEntity = updateOverseasEntity as jest.Mock;
+mockOverseasEntity.mockReturnValue( OVERSEAS_ENTITY_ID );
+
+const mockCloseTransaction = closeTransaction as jest.Mock;
+mockCloseTransaction.mockReturnValue( TRANSACTION_CLOSED_RESPONSE );
+
+const mockPaymentsSession = startPaymentsSession as jest.Mock;
+mockPaymentsSession.mockReturnValue( "CONFIRMATION_URL" );
+
+describe("Update review overseas entity information controller tests", () => {
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("GET tests", () => {
+    test("Thats review statement page is rendered", async () => {
```

Comment:
Is this the default scenario where we're able to view all the existing filing data for the OE? If so can we confirm the following in this test as per the AC:
- Displays contact details
- Displays overseas entity details
- Does not display the BOs section
- Does not display the MOs section

Reply:
We can also confirm the company name/number rendered and there's a 'Continue' button

Reply:
Fixed

Reply:
Thanks!